src/Core/Product/ProductPresenter.php

```diff
@@ -404,8 +404,13 @@ public function addQuantityInformation(
                 if ($product['quantity'] < $settings->lastRemainingItems) {
                     $presentedProduct = $this->applyLastItemsInStockDisplayRule($product, $settings, $presentedProduct);
                 } else {
-                    $presentedProduct['availability_message'] = $product['available_now'];
-                    $presentedProduct['availability'] = 'available';
+                    if($product['quantity_wanted'] > $product['quantity']) {
+                        $presentedProduct['availability_message'] = 'Out of stock';
```

Comment:
It's not really the same thing that "out of stock", it's more something like "we don't have enough products in stock".
Also, you forgot to make the string translatable.